statsmodels/tsa/arima/estimators/gls.py

```diff
@@ -0,0 +1,305 @@
+        include integration and False otherwise.
+    n_iter : int, optional
+        Optionally iterate feasible GSL a specific number of times. Default is
+        to iterate to convergence. If set, this argument overrides the
+        `max_iter` and `tolerance` arguments.
+    max_iter : int, optional
+        Maximum number of feasible GLS iterations. Default is 50. If `n_iter`
+        is set, it overrides this argument.
+    tolerance : float, optional
+        Tolerance for determining convergence of feasible GSL iterations. If
+        `iter` is set, this argument has no effect.
+        Default is 1e-8.
+    arma_estimator : string, optional
+        The estimator used for estimating the ARMA model. This option should
+        not generally be used, unless the default method is failing or is
+        otherwise unsuitable. Possible values are: 'innovations_mle',
+        'statespace', 'hannan_rissanen', 'yule_walker', 'burg', and
+        'innovations'. Not all values will be valid, depending on the
+        specified model orders (`order` and `seasonal_order`).
+        'yule_walker' and 'burg' can only be used with non-seasonal,
+        consecutive autoregressive (AR) models, while 'innovations' can only be
+        used with non-seasonal, consecutive moving average (MA) models.
+        'hannan_rissanen' can be used with any ARMA non-seasonal model.
+        'statespace' and 'innovations_mle' can be used with any
+        specification, including those with seasonal and non-consecutive lag
+        orders. The default is 'innovations_mle'.
+    arma_estimator_kwargs : dict, optional
+        Arguments to pass to the ARMA estimator.
+
+    Returns
+    -------
+    parameters : SARIMAXParams object
+        Contains the parameter estimates from the final iteration.
+    other_results : Bunch
+        Includes eight components: `spec`, `params`, `converged`,
+        `differences`, `iterations`, `arma_estimator`, 'arma_estimator_kwargs',
+        and `arma_results`.
+
+    Notes
+    -----
+    The primary reference is [1]_, section 6.6. In particular, the
+    implementation follows the iterative procedure described in section 6.6.2.
+    Construction of the transformed variables used to compute the GLS estimator
+    described in section 6.6.1 is done via an application of the innovations
+    algorithm (rather than explicit construction of the transformation matrix).
+
+    Note that if the specified model includes integration, both the `endog` and
+    `exog` series will be differenced prior to estimation and a warning will
+    be issued to alert the user.
+
+    TODO: improve "Returns" documentation for `other_results`.
+
+    TODO: Add documentation for why we have e.g. `include_constant` but not
+    other trend specifications (e.g. that it is to maintain consistency with
+    estimation methods with assumptions that require demeaned series) or else
+    revisit this.
+
+    TODO: add support for fixed parameters
```

Comment:
is this TODO for the docstring or the function?

Reply:
I'll move these out and make issues.